File: python/leetcode/009.py
class Solution(object):
    def isPalindrome(self, x):
        if x < 0 or (x % 10 == 0 and x != 0):
            return False
        y = 0
        while y < x:
            y = 10 * y + x % 10
            x /= 10
        return y == x or y / 10 == x

File: cpp/leetcode/009.cpp
class Solution {
public:
    bool isPalindrome(int x) {
        if (x < 0 || (x != 0 && x % 10 == 0)) {
            return false;
        }
        int y = 0;
        while (y < x) {
            y = 10 * y + x % 10;
            x /= 10;
        }
        return y == x || y / 10 == x;
    }
};